src/gc.c

```diff
@@ -2277,10 +2435,51 @@ JL_DLLEXPORT void jl_gc_collect(int full)
 #endif
     if (recollect) {
         n_pause--;
-        jl_gc_collect(0);
+        _jl_gc_collect(0, stack_hi);
     }
 }
 
+JL_DLLEXPORT void jl_gc_collect(int full)
+{
+    if (!is_gc_enabled || jl_in_gc)
+        return;
```

Comment:
No. Multiple threads entering the GC is handled separately below to avoid running the GC multiple times.